modules/data.remote/R/call_MODIS.R

```diff
@@ -14,78 +14,133 @@
-    } else {
-      print("Check #3: Band Exists!")
+    } 
+    
+    #3. check that dates asked for in function parameters are fall within dates available for modis product/bands.
+    dates <- MODISTools::mt_dates(product = product, lat = lat, lon = lon)$modis_date
+    dates <- as.Date(as.character(substr(dates, 2, nchar(dates))), format = "%Y%j")
+    
+    ########## Date case 1:  user only wants one date ##########
+    if (start_date == end_date)
+    {
+      if (as.numeric(start_date) >= dates[1] & as.numeric(end_date) <= dates[length(dates)])
+      {
+        print("Extracting data")
+        
+        start <- as.Date(start_date, "%Y%j")
+        end <- as.Date(end_date, "%Y%j")
+      }
+      ########## For Date case 1: if only one date is asked for, but date is not within modis data prodate date range ##########
+      if (as.numeric(start_date)< dates[1] | as.numeric(end_date)> dates[length(dates)])
+      {
+        print(start)
+        stop("start or end date are not within MODIS data product date range. Please choose another date.")
+      }
+    } else { 
+      ########## Date case 2: user want a range of dates ##########
+      # Best case scenario: Start and end date asked for fall with available date range of modis data product.
+      if (as.numeric(start_date)>=dates[1] & as.numeric(end_date)<=dates[length(dates)])
+      {
+        print("Check #2: All dates are available!")
+      }
+      
+      # Okay scenario: Some MODIS data is available for parameter start_date and end_date range, but either the start_date or end_date falls outside the range of availble 
+      # MODIS data dates
+      if (as.numeric(start_date)<=dates[1] & as.numeric(end_date)>=dates[1])
+      {
+        start_date = dates[1]
+        print("WARNING: Dates are only partially available. Start date before modis data product is available.")
+      } 
+      if (as.numeric(end_date)>=dates[length(dates)] & as.numeric(start_date) <= dates[length(dates)])
+      {
+        end_date = dates[length(dates)]
+        print("WARNING: Dates are only partially available. End date befire modis data product is available.")
+      } 
+      
+      # Unacceptable scenario: start_date and end_date does not fall within the availa MODIS data product date range. There is no data to extract in this scenario.
+      if ((as.numeric(start_date)<dates[1] & as.numeric(end_date <dates[1])) | (as.numeric(start_date)>dates[length(dates)] & as.numeric(end_date)>dates[length(dates)]))
+      {
+        stop("No MODIS data available start_date and end_date parameterized.")
+      }
+      
+      start <- as.Date(start_date, "%Y%j")
+      end <- as.Date(end_date, "%Y%j")
     }
-  
     
     print("Extracting data")
-    
-    start <- as.Date(start_date, "%Y%j")
-    end <- as.Date(end_date, "%Y%j")
+    cat(paste("Product =", product, "\n", "Band =", band, "\n", "Date Range =", start, "-", end, "\n", "Latitude =", lat, "\n", "Longitude =", lon, sep = " "))
```

Comment:
Please use `message` or, better yet, `PEcAn.logger::logger.<something>` here.

Note that `PEcAn.logger::logger.*` functions have a `wrap = FALSE` argument that preserves input formatting, which allows them to be used for printing, e.g. tabular output.